### PR TITLE
Adds support for configuring external_entropy_access on secrets engines

### LINKED ISSUE #765: No option to enable external_entropy_access on secrets engine
Hello

### Terraform Version
0.12.6

### Vault Provider Version
2.10.0

### Affected Resource(s)
vault_mount

### Terraform Configuration Files
```hcl
resource "vault_mount" "example" {
  path        = "transit"
  type        = "transit"
  description = "This is an example mount"
  external_entropy_access = true
}
```

### Debug Output
```
An argument named "external_entropy_access" is not expected here
```

### Expected Behavior
Vault API supports `external_entropy_access`, however there is no option for it in the provider
https://www.vaultproject.io/api-docs/system/mounts

### PR DESCRIPTION
**Pull request by austingebauer**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

Closes #765 

This PR adds support for configuring `external_entropy_access` on secrets engines via the `/sys/mounts/:path` API.

I had to bring both the `vault/api` and `vault/sdk` versions up to https://github.com/hashicorp/vault/commit/47a54ac8bc6c423805bb3be5c462a4676a62ae3c to enable this. See [circleci pipeline](https://app.circleci.com/pipelines/github/terraform-providers/terraform-provider-vault/1059/workflows/2cb0074b-c1f3-4f8e-b4bc-159d15474a3b/jobs/1056/steps) with larger set of tests ran on this change.

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-vault/blob/master/CHANGELOG.md):

```release-note
* For the `/sys/mounts/{path}` endpoint, adds support for `external_entropy_access` ([#792](https://github.com/terraform-providers/terraform-provider-vault/pull/792)).
```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestResourceMount'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test $(go list ./... |grep -v 'vendor') -v -run=TestResourceMount -timeout 120m
?       github.com/terraform-providers/terraform-provider-vault [no test files]
?       github.com/terraform-providers/terraform-provider-vault/cmd/coverage    [no test files]
testing: warning: no tests to run
PASS
ok      github.com/terraform-providers/terraform-provider-vault/util    (cached) [no tests to run]
=== RUN   TestResourceMount
--- PASS: TestResourceMount (0.19s)
=== RUN   TestResourceMount_Local
--- PASS: TestResourceMount_Local (0.17s)
=== RUN   TestResourceMount_SealWrap
--- PASS: TestResourceMount_SealWrap (0.17s)
=== RUN   TestResourceMount_KVV2
--- PASS: TestResourceMount_KVV2 (0.15s)
=== RUN   TestResourceMount_ExternalEntropyAccess
--- PASS: TestResourceMount_ExternalEntropyAccess (0.23s)
PASS
ok      github.com/terraform-providers/terraform-provider-vault/vault   (cached)
...
```

### CODE REVIEW

File: vault/resource_mount.go [hunk 1]
```diff
@@ -92,6 +92,16 @@ func mountResource() *schema.Resource {
...
 				Description: "Enable seal wrapping for the mount, causing values stored by the mount to be wrapped by the seal's encryption capability",
 			},
+
+			"external_entropy_access": {
+				Type:        schema.TypeBool,
+				Required:    false,
+				Optional:    true,
+				Default:     false,
+				ForceNew:    true,
+				Computed:    false,
```
Comment (catsby) [inline vault/resource_mount.go:102]: We should be able to omit `Required` and `Computed` here, they both will receive their default value of  `false`.

File: vault/resource_mount_test.go [hunk 2]
```diff
@@ -140,6 +140,28 @@ func TestResourceMount_KVV2(t *testing.T) {
...
+				Check:  testResourceMount_CheckExternalEntropyAccess(path, false),
+			},
+			{
+				Config: testResourceMount_UpdateConfigExternalEntropyAccess(path, true),
+				Check:  testResourceMount_CheckExternalEntropyAccess(path, true),
+			},
+			{
+				Config: testResourceMount_UpdateConfigExternalEntropyAccess(path, false),
+				Check:  testResourceMount_CheckExternalEntropyAccess(path, false),
+			},
```
Comment (catsby) [inline vault/resource_mount_test.go:160]: Could you add another two `TestStep` entries here, the first using `testResourceMount_UpdateConfigExternalEntropyAccess(path, true),` and the second using `testResourceMount_InitialConfigExternalEntropyAccess(path)`, and verify that `mount.ExternalEntropyAccess` is false? I'd like to ensure that if `external_entropy_access` is initially set to `true`, and then later completely removed from the config (as opposed to set to `false`) that we do the right thing.

Comment (austingebauer) [reply vault/resource_mount_test.go:160]: Good idea! I've added those two additional `TestStep` entries in 51712d7. I also manually tested out going from `external_entropy_access = true` to completely removing it to ensure the configuration on the mount is what's expected (false).

Test rerun:
```
$ TESTARGS='-run=TestResourceMount -count=1' make testacc                     
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test $(go list ./... |grep -v 'vendor') -v -run=TestResourceMount -count=1 -timeout 120m
?       github.com/terraform-providers/terraform-provider-vault [no test files]
?       github.com/terraform-providers/terraform-provider-vault/cmd/coverage    [no test files]
testing: warning: no tests to run
PASS
ok      github.com/terraform-providers/terraform-provider-vault/util    0.471s [no tests to run]
=== RUN   TestResourceMount
--- PASS: TestResourceMount (0.18s)
=== RUN   TestResourceMount_Local
--- PASS: TestResourceMount_Local (0.16s)
=== RUN   TestResourceMount_SealWrap
--- PASS: TestResourceMount_SealWrap (0.16s)
=== RUN   TestResourceMount_KVV2
--- PASS: TestResourceMount_KVV2 (0.13s)
=== RUN   TestResourceMount_ExternalEntropyAccess
--- PASS: TestResourceMount_ExternalEntropyAccess (0.35s)
PASS
ok      github.com/terraform-providers/terraform-provider-vault/vault   1.420s

```

File: vault/resource_mount.go (hunk 1 above)
Comment (austingebauer) [reply vault/resource_mount.go]: Sounds good. I've removed them in fb4facc.

File: vault/resource_mount_test.go (hunk 2 above)
Comment (catsby) [reply vault/resource_mount_test.go:160]: 👍

**Pull request merged**